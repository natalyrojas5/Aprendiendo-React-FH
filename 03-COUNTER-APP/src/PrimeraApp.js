// import React, { Fragment } from 'react';
import React from 'react';
import PropTypes from 'prop-types';

// FUNCTION COMPONENT
const PrimeraApp = ({saludo, subtitulo}) =>{
    return (
        <>
            <h1>{saludo }</h1>
            <p>{subtitulo}</p>
        </>
    );
}

// OBLIGA" A QUE LAS PROPIEDADES TENGAN UN TIPO DEFINIDO
// En el ejemplo "obliga" a que la propiedad *saludo* sea de tipo string, caso contrario mostrar√° un error en la consola

PrimeraApp.propTypes = {
    saludo : PropTypes.string.isRequired
}

// Valores por defecto de las propiedades de mi componente PrimeraApp

PrimeraApp.defaultProps ={
    subtitulo : ''
}

export default PrimeraApp;