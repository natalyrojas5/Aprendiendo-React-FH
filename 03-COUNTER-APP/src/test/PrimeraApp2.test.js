import React from 'react';
import '@testing-library/jest-dom';
import { shallow} from 'enzyme';

import PrimeraApp from '../PrimeraApp';

describe('Pruebas en <PrimeraApp />', ()=>{
    test('Debe de mostrar <PrimerApp /> correctamente"', ()=>{
        const saludo = 'Hola Mundo';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);

        expect(wrapper).toMatchSnapshot();
    });
    test('Debe de mostrar un subtitulo', ()=>{
        const saludo = 'Hola Mundo';
        const subtitulo = 'Hola soy un subtitulo';
        const wrapper = shallow(
            <PrimeraApp 
                saludo={saludo} 
                subtitulo={subtitulo}
            />
        );
        // find() = document.querySelector()
        const parrafo = wrapper.find('p').text();

        expect(parrafo).toBe(subtitulo);
    })
})