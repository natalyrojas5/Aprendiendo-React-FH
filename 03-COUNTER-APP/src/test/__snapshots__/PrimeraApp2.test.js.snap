// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe de mostrar <PrimerApp /> correctamente" 1`] = `
<Fragment>
  <h1>
    Hola Mundo
  </h1>
  <p />
</Fragment>
`;
